compose/cli/main.py

```diff
@@ -84,6 +85,20 @@ def main():
         sys.exit(1)
 
 
+def log_api_error(e):
+    if 'client is newer than server' in e.explanation:
+        # we need JSON formatted errors. In the meantime...
+        client_version = e.explanation.split('client API version: ')[1].split(',')[0]
```

Comment:
Ohh, right!

The refactor I propose in https://github.com/docker/compose/pull/2832#commitcomment-15923800 would make this much easier, because we'd be able to do `command.config.version` to get it.

We also set the client version as `client.api_version`, right?  Maybe we could either add it to a property on `APIError`, or call `docker_client()` to create a new client and use it from there?